subprojects/core-api/src/main/java/org/gradle/api/artifacts/type/ArtifactTypeDefinition.java

```diff
@@ -54,6 +54,30 @@
     @Incubating
     String JVM_RESOURCES_DIRECTORY = "java-resources-directory";
 
+    /**
+     * Represents the packaged C++ public header files
+     *
+     * @since 5.3
+     */
+    @Incubating
+    String C_PLUS_PLUS_API_ZIP_TYPE = "zip";
```

Comment:
Could these be semi-generic?  e.g., this is just a zip vs something C++ specific?

Reply:
Same as above - these are more generic now.